src/ppx/xr_component.cpp

```diff
@@ -663,69 +831,52 @@ bool XrComponent::RemoveLayer(LayerRef layerRef)
+XrCamera& XrComponent::GetCamera()
 {
-    PPX_ASSERT_MSG((mCurrentViewIndex < mViews.size()), "Invalid view index!");
-    const XrView& view = mViews[mCurrentViewIndex];
-    const XrFovf& fov  = view.fov;
-
-    // Save near and far plane values so that they can be referenced
-    // in EndFrame(), as part of the depth layer info submission.
-    // They can only be set once per frame.
-    PPX_ASSERT_MSG(!mNearPlaneForFrame.has_value() || *mNearPlaneForFrame == nearZ, "GetProjectionMatrixForCurrentViewAndSetFrustumPlanes was already called this frame with a different nearZ value.");
-    PPX_ASSERT_MSG(!mFarPlaneForFrame.has_value() || *mFarPlaneForFrame == farZ, "GetProjectionMatrixForCurrentViewAndSetFrustumPlanes was already called this frame with a different farZ value.");
-    mNearPlaneForFrame = nearZ;
-    mFarPlaneForFrame  = farZ;
-
-    const float tan_left  = tanf(fov.angleLeft);
-    const float tan_right = tanf(fov.angleRight);
+    mCurrentViewCamera.UpdateCamera(mViews[mCurrentViewIndex]);
+    return mCurrentViewCamera;
+}
 
-    const float tan_down = tanf(fov.angleDown);
-    const float tan_up   = tanf(fov.angleUp);
+// -------------------------------------------------------------------------------------------------
+// XrCamera
+// -------------------------------------------------------------------------------------------------
 
-    const float tan_width  = tan_right - tan_left;
-    const float tan_height = (tan_up - tan_down);
+void XrCamera::UpdateCamera(const XrView& view)
+{
+    // Static: mPixelAligned, mNearClip, mFarClip
+    // Given: view.pose.orientation, mEyePosition = view.pose.position, view.fov
+    // Solve for: mTarget, mViewDirection, mWorldUp, m*Matrix
 
-    const float a00 = 2 / tan_width;
-    const float a11 = 2 / tan_height;
+    // Calculate projection matrix
+    {
+        const XrFovf& fov = view.fov;
 
-    const float a20 = (tan_right + tan_left) / tan_width;
-    const float a21 = (tan_up + tan_down) / tan_height;
-    const float a22 = -farZ / (farZ - nearZ);
+        const float left  = tanf(fov.angleLeft) * mNearClip;
+        const float right = tanf(fov.angleRight) * mNearClip;
 
-    const float a32 = -(farZ * nearZ) / (farZ - nearZ);
+        const float down = tanf(fov.angleDown) * mNearClip;
+        const float up   = tanf(fov.angleUp) * mNearClip;
 
-    // clang-format off
-    const float mat[16] = {
-        a00,    0,      0,      0,
-        0,      a11,    0,      0,
-        a20,    a21,    a22,    -1,
-        0,      0,      a32,    0,
-    };
-    // clang-format on
+        // Not sure why it is frustumRH_ZO
+        mProjectionMatrix = glm::frustumRH_ZO(left, right, down, up, mNearClip, mFarClip);
+    }
+    // Calculate view
+    {
+        const quat orientation = FromXr(view.pose.orientation);
 
-    return glm::make_mat4(mat);
-}
+        mEyePosition   = FromXr(view.pose.position);
+        mViewDirection = glm::rotate(orientation, float3(0, 0, -1));
```

Comment:
out of curiosity, why is float3(0, 0, -1)?

Reply:
`-Z` is OpenXR forward direction.